core/base/kdTree/KDTree.h

```diff
@@ -64,14 +64,18 @@ namespace ttk {
                     const int &ptNumber,
                     const int &dimension,
                     const std::vector<std::vector<dataType>> &weights = {},
-                    const int weight_number = 1);
+                    const int weight_number = 1,
+                    int nodeNumber = -1);
```

Comment:
`int &nodeNumber = -1` --> `const int &nodeNumber = -1`  (unless it is not read only)